[Woo POS] [Design System]  Spacing and padding consolidation

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #15111

## Description
This PR creates two new struct that we use to hold spacing and padding consolidated values across POS: `POSPadding`, and `POSSpacing`.

I'm waiting for answer regarding xxLarge, as it seems to be the same as xLarge ( 1qcjzXitBHU7xPnpCOWnNM-fi-node-id=21-7403#1141488810 ). I've used `40` for this for now and named it as `xxLarge` on 01ac008b41b5a252ce3b53e9eac2f2e85fc4aadb

<img width="487" alt="Screenshot 2025-02-20 at 13 59 50" src="https://github.com/user-attachments/assets/96761159-fdad-4e9c-9e48-96f0eada6e01" />

I've also removed some warnings coming from NameSpace in previews.

## Testing information

Most changes were just switching the existing value for the consolidated value, some others needed to be changed:

1. Observe that items as products or variations have a `xSmall` spacing between title and the rest: a530615d6b2604fd7c0a3fdafffc6a4933bca209

2. Go through reader connection, and process a payment, observe that "Several readers found" case: f14d59f46e3d0ffe46ecc95bb1eaaeee20f8a36f

![IMG_0634](https://github.com/user-attachments/assets/32f92724-7af9-46db-8b48-ce8771c1149c)

And "Remove card" state still look as always, in consistency with the rest of the flow. This cannot be seen in simulated reader, here's with the physical one (0:37) : f875c8797e6182bb3f7ed74adc9de6500c4734d3 , 5510e49179e2d0b5f21adae6a347b5a4c9f721e2

https://github.com/user-attachments/assets/886b7bc3-43e6-498e-a283-5f0e76a74980

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Define POSPadding struct

iamgabrielma committed:
Define POSSpacing

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/238">21.8</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
apply style to misc views

iamgabrielma committed:
Add xxLarge case

iamgabrielma committed:
Extract styles to their own files

iamgabrielma committed:
Vstack-Hstack updates

iamgabrielma committed:
Adjust FoundMultipleReadersView headerPadding

We have no design for the multiple readers view, reduced the top-bottom padding a bit for consistency.

iamgabrielma committed:
update PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView

iamgabrielma committed:
Update PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageView

iamgabrielma committed:
update cash view conditional padding

iamgabrielma committed:
clean up POSSendReceiptView

iamgabrielma committed:
remove unused

iamgabrielma committed:
Update Vstack text spacing in item card as xSmall

From 1qcjzXitBHU7xPnpCOWnNM/DS-POS?-fi-47-19437&t=B4lkvzVJVJrbmyw3-0 these use gap: spacing-xsmall

staskus committed:
Use POS Design System Constants in CardReaderConnectionStatusView

staskus committed:
Remove unused code in PointOfSaleLoadingView

staskus committed:
Comment out broken 50% button width set for payment failure buttons

staskus committed:
Use POS Design System Constants in Success View

staskus committed:
Use POS Design System Constants in CartView

staskus committed:
Remove unused code in ChildItemList

staskus committed:
Use POS Design System Constants in ItemList

staskus committed:
Use POS Design System Constants in ItemListView

staskus committed:
Use POS Design System Constants in ItemRowView

staskus committed:
Use POS Design System Constants in PaymentButtons

staskus committed:
Use POS Design System Constants in OnboardingView

staskus committed:
Adjust Cash and Email views to use constants

It doesn't fit designs 100% since it looks to use centering between keyboad and header logic. Cash and Email uses different text fields so it's hard to achieve same centering with same constants. Need to use different ones.

staskus committed:
Use POS Design System Constants in Dashboard

staskus committed:
Use POS Design System Constants in Exit Alert

staskus committed:
Remove unnecessary padding in POSNoticeView

staskus committed:
Use POS Design System Constants in POSPageHeaderView

staskus committed:
Use POS Design System Constants and update layout of SimpleProducts modal

staskus committed:
Use POS Design System Constants in TotalsView

staskus committed:
Merge branch 'trunk' into issue/15111-ds-spacing-padding-consolidation

staskus commented:
@iamgabrielma, @jaclync 

I added a bunch of commits with places that were missing `POSSpacing` or `POSPadding`. 

While comparing it with Figma, I also made some adjustments. I tried to make sure I hadn't broken anything and didn't make drastic changes. However, some unexpected issues are possible. I made screenshots of 30 different states just as Android did for easier comparison:

[Feb 20 2024 POS Screenshots.zip](https://github.com/user-attachments/files/18895804/Feb.20.2024.POS.Screenshots.zip)
-----

I also commented out the code that set payment view buttons to 50% since the functionality was broken and need to be fixed https://github.com/woocommerce/woocommerce-ios/pull/15217/commits/19c21ec19219d8c2a5d32166671910f244d9bd53

[[Woo POS][Design System] Update icon colors in card connection and payment flows · Issue #15198 · woocommerce/woocommerce-ios](https://github.com/woocommerce/woocommerce-ios/issues/15198)

jaclync committed:
Merge branch 'trunk' into issue/15111-ds-spacing-padding-consolidation

# Conflicts:
#	WooCommerce/Classes/POS/Presentation/PointOfSaleCollectCashView.swift
#	WooCommerce/Classes/POS/Presentation/Reusable Views/POSSendReceiptView.swift

jaclync commented on WooCommerce/Classes/POS/Utils/POSPadding.swift:
```diff
@@ -0,0 +1,13 @@
+    static let medium: CGFloat = 16
+    static let large: CGFloat = 24
+    static let xLarge: CGFloat = 32
+    static let xxLarge: CGFloat = 40
```
xxLarge in design is 96 now, but it might be too big in our use cases. I'll check as I review.

iamgabrielma replied:
Thanks! I agree that we should keep xxLarge as 40 as is more used, also for consistency between spacing and padding, and add a new one if needed that represents 96, or just an xxxL as 48 and we can do a *2 where needed.

jaclync committed:
Uncomment a possibly unintended change to disable 0.5 width.

jaclync commented on WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift:
```diff
@@ -6,14 +6,14 @@ struct PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView: View {
 
     var body: some View {
-        VStack(alignment: .center, spacing: Layout.headerSpacing) {
+        VStack(alignment: .center, spacing: PointOfSaleCardPresentPaymentLayout.imageAndTextSpacing) {
```
nice 👍

jaclync commented on WooCommerce/Classes/POS/Presentation/POSReceiptEligibilityBanner.swift:
```diff
@@ -28,11 +28,11 @@ struct POSReceiptEligibilityBanner: View {
-        static let imagesize: CGFloat = 40
-        static let imagePadding: CGFloat = 4
-        static let bannerPadding: CGFloat = 16
+        static let imagesize: CGFloat = POSPadding.xxLarge
```
nit: probably better to leave the image size as it is, instead of using the "padding" which is a different concept

jaclync committed:
Revert change to use padding value for image size.

jaclync committed:
Uncomment 0.5 width diffs.

jaclync committed:
Nit: use spacing instead of padding.

jaclync commented on WooCommerce/Classes/POS/Presentation/Order Messages/PointOfSaleOrderSyncErrorMessageView.swift:
```diff
@@ -33,10 +33,10 @@ struct PointOfSaleOrderSyncErrorMessageView: View {
-        static let buttonBottomPadding: CGFloat = 16
+        static let headerSpacing: CGFloat = POSSpacing.large
+        static let textSpacing: CGFloat = POSSpacing.medium
+        static let buttonSidePadding: CGFloat = POSPadding.xxLarge
```
For this POSPadding.xxLarge where the design is 96px now, I'm not sure if it's a good idea. I've asked design about adding 96px to another style like xxxLarge in 1qcjzXitBHU7xPnpCOWnNM-fi-21_7450#1141837081.

jaclync committed:
Use the same vertical spacing for VStack in exit POS view with other tweaks.

jaclync commented on WooCommerce/Classes/POS/Presentation/PointOfSaleExitPosAlertView.swift:
```diff
@@ -4,30 +4,29 @@ import SwiftUI
 struct PointOfSaleExitPosAlertView: View {
     @Environment(\.dismiss) private var dismiss
     @Binding private var isPresented: Bool
-    @Environment(PointOfSaleAggregateModel.self) private var posModel
```
This is to fix preview crash, since posModel isn't used here.

jaclync committed:
Revert card text spacing to use a constant with `POSSpacing.xSmall`.

jaclync commented:
> I also commented out the code that set payment view buttons to 50% since the functionality was broken and need to be fixed [19c21ec](https://github.com/woocommerce/woocommerce-ios/commit/19c21ec19219d8c2a5d32166671910f244d9bd53)

Just saw this, I was wondering why these were commented out and thought it was an accident. I'll test this to see how we can fix it.

jaclync committed:
Use padding for cart checkout button vertical padding, update empty cart font to match design.

jaclync commented on WooCommerce/Classes/POS/Presentation/Reusable Views/POSNoticeView.swift:
```diff
@@ -58,7 +58,6 @@ struct POSNoticeView<Content: View>: View {
                                 .accessibilityLabel(Localization.dismissAccessibilityLabel)
                         }
                     )
-                    .padding(Constants.dismissIconPadding)
```
From the design where X icon is a 32x32 container while the icon font size is 20px, I believe this 6px padding is necessary. I'm reverting this change for now, lemme know if I misunderstood anything.

jaclync committed:
Revert custom dismiss icon padding in notice view.

jaclync commented on WooCommerce/Classes/POS/Presentation/Reusable Views/POSPageHeaderView.swift:
```diff
@@ -95,9 +95,9 @@ struct POSPageHeaderView<TrailingContent: View>: View {
-    static let backButtonHorizontalPadding: CGFloat = 12
-    static let horizontalSpacing: CGFloat = 16
-    static let titleSubtitleSpacing: CGFloat = 4
+    static let backButtonHorizontalPadding: CGFloat = POSSpacing.xSmall + POSSpacing.small
```
Similar to the notice view custom X icon padding, this is from deducting the icon width ~24px from the icon container 48x48 (the icon dimension varies). I'll update the value to 12 with a comment.

jaclync committed:
Revert back button horizontal padding to a custom value with a comment about how it is calculated.

jaclync committed:
Tweak spacing/padding and font/color in `SimpleProductsOnlyInformation`.

jaclync committed:
Remove unused code in success view.

jaclync committed:
Tweak spacing in totals view to be closer to design which is using spacing not in the design system.

jaclync committed:
Tweak design in cash view.

jaclync committed:
Tweak design in receipt view to be consistent with cash view.

jaclync commented on WooCommerce/Classes/POS/Presentation/Reusable Views/POSSendReceiptView.swift:
```diff
@@ -34,31 +34,35 @@ struct POSSendReceiptView: View {
-                            .font(POSFontStyle.posBodySmallRegular())
-                            .foregroundColor(.posError)
-                            .padding(.bottom, Constants.errorMessagePadding)
+                    VStack(alignment: .center, spacing: POSSpacing.xSmall) {
```
I've moved the text field + optional error message to a VStack with xSmall spacing as in the cash view.

jaclync committed:
Disable the 0.5x width button modifier until the parent container width is fixed later.

jaclync commented:
> > I also commented out the code that set payment view buttons to 50% since the functionality was broken and need to be fixed [19c21ec](https://github.com/woocommerce/woocommerce-ios/commit/19c21ec19219d8c2a5d32166671910f244d9bd53)
> 
> Just saw this, I was wondering why these were commented out and thought it was an accident. I'll test this to see how we can fix it.

@staskus Could you share your steps to repro the 50% width issue? Is this the issue - the button isn't filling 50% of the screen width, like in the screenshot below? If so, it's because the container isn't filling the parent width probably with some padding. The padding logic is a bit tricky as there are many states in `TotalsView`. I'll fix this separately, and removed these modifiers for now in https://github.com/woocommerce/woocommerce-ios/pull/15217/commits/795a764ea1f76513d7698b185f32960a54775c69 to enable merge for the release.

error from invalid order amount e.g. .65 | ui hierarchy
-- | --
![Simulator Screenshot - iPad mini (A17 Pro) - 2025-02-21 at 13 37 29](https://github.com/user-attachments/assets/6d432758-d4dc-4357-b71a-9976d0a5d077) | <img width="776" alt="Screenshot 2025-02-21 at 1 40 08 PM" src="https://github.com/user-attachments/assets/e75dfa50-fe94-44f2-ab58-4e9afb77d85d" />

jaclync reviewed (approved):
Nice team work @iamgabrielma @staskus! I went through all the affected views that aren't a straightforward value replacement, and added a few commits while comparing with the latest design along the way. I'm doing another round of testing and enabling auto-merge for the code freeze now.
